### PR TITLE
Set little_endian in the same way that boolbv_byte_extract already does

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 7bb7e529f9bc6b5b7ca4a051bdfe9e317500601d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77137942

File: src/solvers/flattening/boolbv_byte_update.cpp [hunk 1]
```diff
@@ -25,14 +25,7 @@ bvt boolbvt::convert_byte_update(const byte_update_exprt &expr)
...
 
-  bool little_endian;
-
-  if(expr.id()==ID_byte_update_little_endian)
-    little_endian=true;
-  else if(expr.id()==ID_byte_update_big_endian)
-    little_endian=false;
-  else
-    UNREACHABLE;
+  const bool little_endian = expr.id() == ID_byte_update_little_endian;
```
Comment (peterschrammel) [inline src/solvers/flattening/boolbv_byte_update.cpp]: I assume the invariant that `id()` can only have two values here has already been checked higher up.

**Review by peterschrammel — APPROVED** [review]

File: src/solvers/flattening/boolbv_byte_update.cpp (hunk 1 above)
Comment (tautschnig) [reply src/solvers/flattening/boolbv_byte_update.cpp]: I have added a `PRECONDITION` to that effect.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 34404f0db938b954408154459a41c530023f55b0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78386515

**Pull request merged**